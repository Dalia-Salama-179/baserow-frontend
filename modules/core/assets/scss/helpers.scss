.clearfix::after {
  clear: both;
  content: '';
  display: table;
}

.hidden {
  display: none;
}

.visibility-hidden {
  visibility: hidden;
  transform: translateX(-10000px) translateY(-10000px);
}

.align-right {
  text-align: right;
}

.background-white {
  background-color: $white;
}

.color-primary {
  color: $color-primary-500 !important;
}

.color-success {
  color: $color-success-500 !important;
}

.color-warning {
  color: $color-warning-500 !important;
}

.color-error {
  color: $color-error-500 !important;
}

.margin-top-0 {
  margin-top: 0 !important;
}

.margin-top-1 {
  margin-top: 8px !important;
}

.margin-top-2 {
  margin-top: 16px !important;
}

.margin-top-3 {
  margin-top: 30px !important;
}

.margin-top-4 {
  margin-top: 40px !important;
}

.margin-bottom-0 {
  margin-bottom: 0 !important;
}

.margin-bottom-1 {
  margin-bottom: 8px !important;
}

.margin-bottom-2 {
  margin-bottom: 16px !important;
}

.margin-bottom-3 {
  margin-bottom: 30px !important;
}

.margin-bottom-4 {
  margin-bottom: 40px !important;
}

.margin-left-auto {
  margin-left: auto !important;
}

.margin-left-0 {
  margin-left: 0 !important;
}

.margin-left-1 {
  margin-left: 8px !important;
}

.margin-left-2 {
  margin-left: 16px !important;
}

.margin-right-auto {
  margin-right: auto !important;
}

.margin-right-0 {
  margin-right: 0 !important;
}

.margin-right-1 {
  margin-right: 8px !important;
}

.margin-right-2 {
  margin-right: 16px !important;
}

.resizing-horizontal {
  cursor: col-resize;
}

// This selector can only be used to temporarily show an element for calculating
// positions.
.forced-block {
  display: block !important;
}

// Sometimes we forcefully want to disable the the user-select property to initial
// because the element is for example being edited via a contenteditable attribute
// which does not work in combination with the user-select: none; property in Safari.
.forced-user-select-initial {
  user-select: initial !important;
}

.forced-text-overflow-initial {
  text-overflow: initial !important;
}

.prevent-scroll {
  overflow: hidden !important;
}

@keyframes spin {
  0% {
    transform: rotate(0);
  }

  50% {
    transform: rotate(360deg);
  }

  100% {
    transform: rotate(720deg);
  }
}
