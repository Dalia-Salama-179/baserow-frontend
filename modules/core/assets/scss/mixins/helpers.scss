@mixin fixed-height($height, $font-size) {
  height: $height;
  line-height: $height;
  font-size: $font-size;
}

@mixin center-text($width, $font-size, $height: $width) {
  @include fixed-height($height, $font-size);

  width: $width;
  text-align: center;
}

@mixin loading($size: 1.4rem) {
  position: relative;
  display: block;
  width: $size;
  height: $size;
  border-radius: 50%;
  border: 0.25em solid;
  border-color: $color-primary-500 transparent $color-primary-500 transparent;
  animation: spin infinite 1800ms;
  animation-timing-function: cubic-bezier(0.785, 0.135, 0.15, 0.86);
}

// The basically works the same as padding: 10px 0 0 10px or padding: 10px; only then
// resulting position: absolute; with corresponding top, right, bottom, left values.
// -------
// Example:
//  @include absolute(10px);
// Would result in:
//  position: absolute;
//  top: 10px; right: 10px; bottom: 10px; left: 10px;
// -------
// Example:
//  @include absolute(10px, 20px);
// Would result in:
//  position: absolute;
//  top: 10px; right: 20px; bottom: 10px; left: 20px;
// -------
// Example:
//  @include absolute(10px, 20px, 30px);
// Would result in:
//  position: absolute;
//  top: 10px; right: 20px; bottom: 30px;
@mixin absolute($top: null, $right: null, $bottom: null, $left: null) {
  position: absolute;

  @if ($top != null and $right == null and $bottom == null and $left == null) {
    top: $top;
    right: $top;
    bottom: $top;
    left: $top;
  }

  /* stylelint-disable-next-line at-rule-no-unknown */
  @else if ($top != null and $right != null and $bottom == null and
  $left == null) {
    top: $top;
    right: $right;
    bottom: $top;
    left: $right;
  }

  /* stylelint-disable-next-line at-rule-no-unknown */
  @else {
    @if $top != null {
      top: $top;
    }

    @if $right != null {
      right: $right;
    }

    @if $bottom != null {
      bottom: $bottom;
    }

    @if $left != null {
      left: $left;
    }
  }
}
