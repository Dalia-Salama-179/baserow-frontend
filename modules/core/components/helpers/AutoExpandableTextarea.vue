<template>
  <textarea
    ref="inputTextArea"
    :value="value"
    :placeholder="placeholder"
    class="input auto-expandable-textarea"
    :style="{
      height: textBoxSize + 'px',
      overflow: textBoxOverflow,
    }"
    rows="1"
    @input="$emit('input', $event.target.value)"
    @click="$emit('click', $event)"
    @keyup="$emit('keyup', $event)"
    @keydown="$emit('keydown', $event)"
    @blur="$emit('blur', $event)"
  />
</template>
<script>
/**
 * A blank textarea with no surrounding box styling (see AutoExpandableTextareaInput if
 * you want a nice styled box around this) which will automatically expand given the
 * users input to the maximum number of rows specified by the maxRows prop.
 */
export default {
  name: 'AutoExpandableTextarea',
  props: {
    value: {
      required: true,
      type: String,
    },
    loading: {
      required: false,
      default: false,
      type: Boolean,
    },
    placeholder: {
      required: false,
      type: String,
      default: '',
    },
    maxRows: {
      required: false,
      type: Number,
      default: 4,
    },
    startingRows: {
      required: false,
      type: Number,
      default: 1,
    },
  },
  data() {
    return {
      numTextAreaLines: this.startingRows,
    }
  },
  computed: {
    textBoxSize() {
      return 22 * Math.min(this.numTextAreaLines, this.maxRows)
    },
    textBoxOverflow() {
      return this.numTextAreaLines > this.maxRows ? 'auto' : 'hidden'
    },
  },
  watch: {
    value() {
      this.resizeTextArea()
    },
  },
  mounted() {
    this.resizeTextArea()
  },
  methods: {
    resizeTextArea() {
      this.$nextTick(() => {
        const inputTextArea = this.$refs.inputTextArea
        this.numTextAreaLines = this.calculateHeight(inputTextArea)
      })
    },
    /**
     * Taken from https://stackoverflow.com/questions/1760629/how-to-get-number-of-rows-in-textarea-using-javascript/1761203#1761203
     *
     * The key reason we need this is to resize a fully expanded textarea to something
     * smaller as a user deletes newlines or text. Hence we need to actually manipulate
     * the dom and lower the height of the textarea until we find overflow is occurring
     * again to re-find the correct min height.
     */
    calculateContentHeight(ta, scanAmount) {
      const origHeight = ta.style.height
      let height = ta.offsetHeight
      const scrollHeight = ta.scrollHeight
      const overflow = ta.style.overflow
      /// only bother if the ta is bigger than content
      if (height >= scrollHeight) {
        /// check that our browser supports changing dimension
        /// calculations mid-way through a function call...
        ta.style.height = height + scanAmount + 'px'
        /// because the scrollbar can cause calculation problems
        ta.style.overflow = 'hidden'
        /// by checking that scrollHeight has updated
        if (scrollHeight < ta.scrollHeight) {
          /// now try and scan the ta's height downwards
          /// until scrollHeight becomes larger than height
          while (ta.offsetHeight >= ta.scrollHeight) {
            ta.style.height = (height -= scanAmount) + 'px'
          }
          /// be more specific to get the exact height
          while (ta.offsetHeight < ta.scrollHeight) {
            ta.style.height = height++ + 'px'
          }
          height--
          /// reset the ta back to it's original height
          ta.style.height = origHeight
          /// put the overflow back
          ta.style.overflow = overflow
          return height
        } else {
          // We weren't able to resize this element inside this function call to figure
          // out it's true height so we give up and just return the scrollHeight.
          return scrollHeight
        }
      } else {
        return scrollHeight
      }
    },
    calculateHeight(ta) {
      const style = window.getComputedStyle
        ? window.getComputedStyle(ta)
        : ta.currentStyle

      // This will get the line-height only if it is set in the css,
      // otherwise it's "normal"
      const taLineHeight = parseInt(style.lineHeight, 10)
      // Get the scroll height of the textarea
      const taHeight = this.calculateContentHeight(ta, taLineHeight)
      // calculate the number of lines
      return Math.ceil(taHeight / taLineHeight)
    },
  },
}
</script>
