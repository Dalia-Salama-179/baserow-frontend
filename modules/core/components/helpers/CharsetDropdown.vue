<template>
  <Dropdown :value="value" :disabled="disabled" @input="$emit('input', $event)">
    <DropdownItem name="Unicode (UTF-8)" value="utf-8"></DropdownItem>
    <DropdownItem name="Arabic (ISO-8859-6)" value="iso-8859-6"></DropdownItem>
    <DropdownItem
      name="Arabic (Windows-1256)"
      value="windows-1256"
    ></DropdownItem>
    <DropdownItem name="Baltic (ISO-8859-4)" value="iso-8859-4"></DropdownItem>
    <DropdownItem
      name="Baltic (windows-1257)"
      value="windows-1257"
    ></DropdownItem>
    <DropdownItem
      name="Celtic (ISO-8859-14)"
      value="iso-8859-14"
    ></DropdownItem>
    <DropdownItem
      name="Central European (ISO-8859-2)"
      value="iso-8859-2"
    ></DropdownItem>
    <DropdownItem
      name="Central European (Windows-1250)"
      value="windows-1250"
    ></DropdownItem>
    <DropdownItem name="Chinese, Simplified (GBK)" value="gbk"></DropdownItem>
    <DropdownItem name="Chinese (GB18030)" value="gb18030"></DropdownItem>
    <DropdownItem name="Chinese Traditional (Big5)" value="big5"></DropdownItem>
    <DropdownItem name="Cyrillic (KOI8-R)" value="koi8-r"></DropdownItem>
    <DropdownItem name="Cyrillic (KOI8-U)" value="koi8-u"></DropdownItem>
    <DropdownItem
      name="Cyrillic (ISO-8859-5)"
      value="iso-8859-5"
    ></DropdownItem>
    <DropdownItem
      name="Cyrillic (Windows-1251)"
      value="windows-1251"
    ></DropdownItem>
    <DropdownItem
      name="Cyrillic Mac OS (x-mac-cyrillic)"
      value="x-mac-cyrillic"
    ></DropdownItem>
    <DropdownItem name="Greek (ISO-8859-7)" value="iso-8859-7"></DropdownItem>
    <DropdownItem
      name="Greek (Windows-1253)"
      value="windows-1253"
    ></DropdownItem>
    <DropdownItem name="Hebrew (ISO-8859-8)" value="iso-8859-8"></DropdownItem>
    <DropdownItem
      name="Hebrew (Windows-1255)"
      value="windows-1255"
    ></DropdownItem>
    <DropdownItem name="Japanese (EUC-JP)" value="euc-jp"></DropdownItem>
    <DropdownItem
      name="Japanese (ISO-2022-JP)"
      value="iso-2022-jp"
    ></DropdownItem>
    <DropdownItem name="Japanese (Shift-JIS)" value="shift-jis"></DropdownItem>
    <DropdownItem name="Korean (EUC-KR)" value="euc-kr"></DropdownItem>
    <DropdownItem name="Macintosh" value="macintosh"></DropdownItem>
    <DropdownItem
      name="Nordic (ISO-8859-10)"
      value="iso-8859-10"
    ></DropdownItem>
    <DropdownItem
      name="South-Eastern European (ISO-8859-16)"
      value="iso-8859-16"
    ></DropdownItem>
    <DropdownItem name="Thai (Windows-874)" value="windows-874"></DropdownItem>
    <DropdownItem
      name="Turkish (Windows-1254)"
      value="windows-1254"
    ></DropdownItem>
    <DropdownItem
      name="Vietnamese (Windows-1258)"
      value="windows-1258"
    ></DropdownItem>
    <DropdownItem
      name="Western European (ISO-8859-1)"
      value="iso-8859-1"
    ></DropdownItem>
    <DropdownItem
      name="Western European (Windows-1252)"
      value="windows-1252"
    ></DropdownItem>
    <DropdownItem name="Latin 3 (ISO-8859-3)" value="iso-8859-3"></DropdownItem>
  </Dropdown>
</template>

<script>
export default {
  // Please keep in sync with
  // src/baserow/contrib/database/api/export/serializers.py:SUPPORTED_EXPORT_CHARSETS
  name: 'CharsetDropdown',
  props: {
    value: {
      type: String,
      required: false,
      default: null,
    },
    disabled: {
      type: Boolean,
      required: false,
      default: false,
    },
  },
}
</script>
