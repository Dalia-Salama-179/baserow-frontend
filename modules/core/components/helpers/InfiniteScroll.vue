<template>
  <section
    ref="infiniteScroll"
    :class="{ 'infinite-scroll--reversed': reverse }"
    class="infinite-scroll"
    @scroll="handleScroll"
    v-on="$listeners"
  >
    <slot />
    <div
      v-show="currentCount < maxCount"
      ref="loadingWrapper"
      class="infinite-scroll__loading-wrapper"
    >
      <div v-if="loading" class="loading"></div>
    </div>
    <slot
      v-if="renderEnd && currentCount >= maxCount && wrapperHasScrollbar"
      name="end"
    >
      <div class="infinite-scroll__end-line"></div>
    </slot>
  </section>
</template>

<script>
/**
 * The InfiniteScroll component is designed to wrap a list paginated list of items.
 * Provide it with the currentCount of loaded items and the maxCount that is available
 * from the server and it will emit 'load-next-page' events when the user scrolls to
 * the end of the current items.
 *
 * If reverse is set then the list of items is rendered bottom to top and when the user
 * scrolls to the top the next page is loaded.
 *
 * Provide an end slot which will be shown when the user scrolls to the end of all
 * available items.
 */
export default {
  props: {
    currentCount: {
      type: Number,
      required: true,
    },
    maxCount: {
      type: Number,
      required: true,
    },
    pageSize: {
      type: Number,
      required: false,
      // Please keep in sync with the default page size in
      // backend/src/baserow/api/pagination.py
      default: 100,
    },
    loading: {
      type: Boolean,
      required: true,
    },
    reverse: {
      type: Boolean,
      required: false,
      default: false,
    },
    renderEnd: {
      type: Boolean,
      required: false,
      default: true,
    },
  },
  data() {
    return {
      // We don't want to show the end-line unless the user has scrolled somewhere and
      // hit the end. This also helps us properly align any end-line as it always has
      // a scrollbar and can be calculated as such.
      wrapperHasScrollbar: false,
    }
  },
  watch: {
    currentCount() {
      this.calculateIfWrapperHasScrollbar()
    },
  },
  created() {
    if (this.reverse) {
      // In reverse mode the start is actually the very bottom of the wrapper, so we
      // need to scroll there immediately.
      this.$nextTick(() => {
        this.scrollToStart()
      })
    }
    this.calculateIfWrapperHasScrollbar()
  },
  methods: {
    calculateIfWrapperHasScrollbar() {
      this.$nextTick(() => {
        const infiniteScroll = this.$refs.infiniteScroll
        this.wrapperHasScrollbar =
          infiniteScroll &&
          infiniteScroll.scrollHeight > infiniteScroll.clientHeight
      })
    },
    /**
     * Action listener called when the scroll wrapper scrolls, triggers a load next
     * page event if the user has scrolled to the end of the wrapper.
     */
    handleScroll({ target: { scrollTop, clientHeight, scrollHeight } }) {
      const height = clientHeight + this.$refs.loadingWrapper.clientHeight
      if (this.reverse) {
        if (-scrollTop + height >= scrollHeight) {
          this.loadNextPage()
        }
      } else if (scrollTop + height >= scrollHeight) {
        this.loadNextPage()
      }
    },
    /**
     * Emits a load-next-page event if there are still things to load in from the server
     * and we aren't already loading.
     */
    loadNextPage() {
      if (this.currentCount < this.maxCount && !this.loading) {
        const nextPage = Math.ceil(this.currentCount / this.pageSize) + 1
        this.$emit('load-next-page', nextPage)
      }
    },
    scrollToStart() {
      const infiniteScroll = this.$refs.infiniteScroll
      infiniteScroll.scrollTop = this.reverse ? 0 : infiniteScroll.scrollHeight
    },
  },
}
</script>
