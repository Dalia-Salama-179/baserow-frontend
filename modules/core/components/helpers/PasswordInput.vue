<!--
This helper component can only be used in a form which uses the password validation
from modules/core/validators.js
as ist expects the following vuelidate validation state params

- required
- maxLength
- minLength

In order to use this component in another component/form it needs a value
prop which can be passed via the 'v-model' directive. That means the state for the 
password value always lives in the parent component.
The parent component also needs to provide the vuelidate password validation in order
for this component to show all the possible validation errors.

Example Usage:

<PasswordInput
   v-model="PASSWORDSTATE"
   :validation-state="the vuelidate validation instance of the given password value"
/>

In case the password validation rules change, the passwordValidation object in the
validators file needs to be updated, as well as this component.

modules/core/validators.js

-->
<template>
  <div>
    <input
      :class="{ 'input--error': validationState.$error }"
      type="password"
      class="input input--large"
      :value="value"
      @blur="validationState.$touch()"
      @input="$emit('input', $event.target.value)"
    />
    <div
      v-if="validationState.$error && !validationState.required"
      class="error"
    >
      {{ $t('error.inputRequired') }}
    </div>
    <div
      v-if="validationState.$error && !validationState.maxLength"
      class="error"
    >
      {{
        $t('error.maxLength', {
          max: validationState.$params.maxLength.max,
        })
      }}
    </div>
    <div
      v-if="validationState.$error && !validationState.minLength"
      class="error"
    >
      {{
        $t('error.minLength', {
          min: validationState.$params.minLength.min,
        })
      }}
    </div>
  </div>
</template>

<script>
export default {
  name: 'PasswordInput',
  props: {
    validationState: {
      type: Object,
      required: true,
    },
    value: {
      type: String,
      required: true,
    },
  },
}
</script>
