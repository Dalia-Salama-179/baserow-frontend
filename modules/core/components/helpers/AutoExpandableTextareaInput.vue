<template>
  <div
    class="auto-expandable-textarea__container"
    :class="{
      'auto-expandable-textarea--loading-overlay': loading,
    }"
  >
    <div v-if="loading" class="auto-expandable-textarea--loading"></div>
    <AutoExpandableTextarea
      v-model="innerValue"
      :placeholder="placeholder"
      :max-rows="maxRows"
      :starting-rows="startingRows"
      @keydown.enter.exact.prevent="$emit('entered')"
    >
    </AutoExpandableTextarea>
  </div>
</template>
<script>
import AutoExpandableTextarea from '@baserow/modules/core/components/helpers/AutoExpandableTextarea'

/**
 * A textarea styled as an input box which will automatically expand given the users
 * input to the maximum number of rows specified by the maxRows prop. Prevents enter
 * by default and instead emits an 'entered' event for the parent to handle. Also shows
 * a loading overlay when the loading prop is set to true.
 */
export default {
  name: 'AutoExpandableTextareaInput',
  components: {
    AutoExpandableTextarea,
  },
  props: {
    value: {
      required: true,
      type: String,
    },
    loading: {
      required: false,
      default: false,
      type: Boolean,
    },
    placeholder: {
      required: false,
      type: String,
      default: '',
    },
    maxRows: {
      required: false,
      type: Number,
      default: 4,
    },
    startingRows: {
      required: false,
      type: Number,
      default: 1,
    },
  },
  computed: {
    innerValue: {
      get() {
        return this.value
      },
      set(value) {
        this.$emit('input', value)
      },
    },
  },
}
</script>
