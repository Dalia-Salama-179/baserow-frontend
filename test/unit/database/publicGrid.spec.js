import { TestApp } from '@baserow/test/helpers/testApp'
import PublicGrid from '@baserow/modules/database/pages/publicGridView'

// Mock out debounce so we dont have to wait or simulate waiting for the various
// debounces in the search functionality.
jest.mock('lodash/debounce', () => jest.fn((fn) => fn))

describe('Public View Page Tests', () => {
  let testApp = null
  let mockServer = null

  beforeAll(() => {
    testApp = new TestApp()
    mockServer = testApp.mockServer
  })

  afterEach(() => testApp.afterEach())

  test('Can see a publicly shared grid view', async () => {
    const slug = 'testSlug'
    const gridViewName = 'my public grid view name'
    givenAPubliclySharedGridViewWithSlug(gridViewName, slug)

    const publicGridViewPage = await testApp.mount(PublicGrid, {
      asyncDataParams: {
        slug,
      },
    })

    expect(publicGridViewPage.html()).toContain(gridViewName)
    expect(publicGridViewPage.element).toMatchSnapshot()
  })

  function givenAPubliclySharedGridViewWithSlug(name, slug) {
    const fields = [
      {
        id: 0,
        name: 'Name',
        type: 'text',
        primary: true,
      },
      {
        id: 1,
        name: 'Last name',
        type: 'text',
      },
      {
        id: 2,
        name: 'Notes',
        type: 'long_text',
      },
      {
        id: 3,
        name: 'Active',
        type: 'boolean',
      },
    ]
    const gridView = mockServer.createPublicGridView(slug, {
      name,
      fields,
    })
    mockServer.createPublicGridViewRows(slug, fields, [
      {
        id: 1,
        order: 0,
        field_1: 'name',
        field_2: 'last_name',
        field_3: 'notes',
        field_4: false,
      },
    ])

    return { gridView }
  }
})
