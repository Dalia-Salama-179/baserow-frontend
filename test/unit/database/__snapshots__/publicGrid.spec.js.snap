// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Public View Page Tests Can see a publicly shared grid view 1`] = `
<div>
  <div
    class="notifications"
  >
    <div
      class="top-right-notifications"
    >
      <!---->
       
      <!---->
       
    </div>
     
    <div
      class="bottom-right-notifications"
    >
      <!---->
       
    </div>
  </div>
   
  <div
    class="grid-view__public-shared-page"
  >
    <div>
      <header
        class="layout__col-2-1 header"
      >
        <div
          class="header__loading"
          style="display: none;"
        />
         
        <ul
          class="header__filter"
        >
          <li
            class="header__filter-item header__filter-item--grids"
          >
            <a
              class="header__filter-link header__filter-link--disabled"
            >
              <i
                class="header__filter-icon header-filter-icon--view fas fa-fw color-primary fa-bars"
              />
               
              <span
                class="header__filter-name header__filter-name--forced"
              >
                <span
                  class=""
                  contenteditable="false"
                >
                  my public grid view name
                </span>
              </span>
            </a>
             
            <!---->
          </li>
           
          <!---->
           
          <li
            class="header__filter-item"
          >
            <div>
              <a
                class="header__filter-link"
              >
                <i
                  class="header__filter-icon fas fa-filter"
                />
                 
                <span
                  class="header__filter-name"
                >
                  viewFilter.filter - 0
                </span>
              </a>
               
            </div>
          </li>
           
          <li
            class="header__filter-item"
          >
            <div>
              <a
                class="header__filter-link"
              >
                <i
                  class="header__filter-icon fas fa-sort"
                />
                 
                <span
                  class="header__filter-name"
                >
                  viewSort.sort - 0
                </span>
              </a>
               
            </div>
          </li>
           
          <!---->
        </ul>
         
        <ul
          class="header__filter header__filter--full-width"
          database="[object Object]"
          table="[object Object]"
        >
          <li
            class="header__filter-item"
          >
            <div>
              <a
                class="header__filter-link"
              >
                <i
                  class="header__filter-icon fas fa-eye-slash"
                />
                 
                <span
                  class="header__filter-name"
                >
                  gridViewHide.hideField - 0
                </span>
              </a>
               
            </div>
          </li>
           
          <li
            class="header__filter-item header__filter-item--right"
          >
            <div>
              <a
                class="header__filter-link"
              >
                <i
                  class="header__search-icon fas fa-search"
                />
                
    
  
              </a>
               
            </div>
          </li>
        </ul>
      </header>
       
      <div
        class="layout__col-2-2 content"
      >
        <div
          class="grid-view"
        >
          <div
            class="scrollbars"
            style="left: 260px;"
          >
            <!---->
             
            <!---->
          </div>
           
          <div
            class="grid-view__left"
            style="width: 260px;"
          >
            <div
              class="grid-view__inner"
              style="min-width: 260px;"
            >
              <div
                class="grid-view__head"
              >
                <div
                  class="grid-view__column"
                  style="width: 60px;"
                />
                 
                <div
                  class="grid-view__column"
                  filters=""
                  style="width: 200px;"
                >
                  <div
                    class="grid-view__description"
                  >
                    <div
                      class="grid-view__description-icon"
                    >
                      <i
                        class="fas fa-font"
                      />
                    </div>
                     
                    <div
                      class="grid-view__description-name"
                    >
                      
      Name
    
                    </div>
                     
                    <!---->
                     
                    <!---->
                     
                    <!---->
                     
                    <!---->
                  </div>
                </div>
                 
                <!---->
              </div>
               
              <div
                class="grid-view__body"
              >
                <div
                  class="grid-view__body-inner"
                >
                  <div
                    class="grid-view__placeholder"
                    style="height: 33px; width: 260px;"
                  >
                    <div
                      class="grid-view__placeholder-column"
                      style="left: 199px;"
                    />
                  </div>
                   
                  <div
                    class="grid-view__rows"
                    style="transform: translateY(0px) translateX(0px);"
                  >
                    <div
                      class="grid-view__row"
                    >
                      <!---->
                       
                      <div
                        class="grid-view__column"
                        style="width: 60px;"
                      >
                        <div
                          class="grid-view__row-info"
                        >
                          <div
                            class="grid-view__row-count"
                            title="1"
                          >
                            
          1
        
                          </div>
                           
                          <!---->
                           
                          <a
                            class="grid-view__row-more"
                          >
                            <i
                              class="fas fa-expand"
                            />
                          </a>
                        </div>
                      </div>
                       
                      <div
                        class="grid-view__column"
                        style="width: 200px;"
                      >
                        <div
                          class="grid-view__cell"
                        >
                          <div
                            class="grid-field-text"
                          >
                            
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                   
                  <!---->
                </div>
              </div>
               
              <div
                class="grid-view__foot"
              >
                <div
                  class="grid-view__foot-info"
                >
                  
        gridView.rowCount - 1
      
                </div>
                 
                <!---->
              </div>
            </div>
             
            <div
              style="display: none;"
            >
              <div
                class="grid-view__field-dragging"
                style="width: 0px; left: 0px;"
              />
               
              <div
                class="grid-view__field-target"
                style="left: 0px;"
              />
            </div>
          </div>
           
          <div
            class="grid-view__divider"
            style="left: 260px;"
          />
           
          <div
            class="grid-view__divider-width"
            style="left: 260px;"
          />
           
          <div
            class="grid-view__right"
            style="left: 260px;"
          >
            <div
              class="grid-view__inner"
              style="min-width: 800px;"
            >
              <div
                class="grid-view__head"
              >
                <!---->
                 
                <div
                  class="grid-view__column"
                  filters=""
                  style="width: 200px;"
                >
                  <div
                    class="grid-view__description"
                  >
                    <div
                      class="grid-view__description-icon"
                    >
                      <i
                        class="fas fa-font"
                      />
                    </div>
                     
                    <div
                      class="grid-view__description-name"
                    >
                      
      Last name
    
                    </div>
                     
                    <!---->
                     
                    <!---->
                     
                    <!---->
                     
                    <div
                      class="grid-view__description-width"
                    />
                  </div>
                </div>
                <div
                  class="grid-view__column"
                  filters=""
                  style="width: 200px;"
                >
                  <div
                    class="grid-view__description"
                  >
                    <div
                      class="grid-view__description-icon"
                    >
                      <i
                        class="fas fa-align-left"
                      />
                    </div>
                     
                    <div
                      class="grid-view__description-name"
                    >
                      
      Notes
    
                    </div>
                     
                    <!---->
                     
                    <!---->
                     
                    <!---->
                     
                    <div
                      class="grid-view__description-width"
                    />
                  </div>
                </div>
                <div
                  class="grid-view__column"
                  filters=""
                  style="width: 200px;"
                >
                  <div
                    class="grid-view__description"
                  >
                    <div
                      class="grid-view__description-icon"
                    >
                      <i
                        class="fas fa-check-square"
                      />
                    </div>
                     
                    <div
                      class="grid-view__description-name"
                    >
                      
      Active
    
                    </div>
                     
                    <!---->
                     
                    <!---->
                     
                    <!---->
                     
                    <div
                      class="grid-view__description-width"
                    />
                  </div>
                </div>
                 
                <!---->
              </div>
               
              <div
                class="grid-view__body"
              >
                <div
                  class="grid-view__body-inner"
                >
                  <div
                    class="grid-view__placeholder"
                    style="height: 33px; width: 600px;"
                  >
                    <div
                      class="grid-view__placeholder-column"
                      style="left: 199px;"
                    />
                    <div
                      class="grid-view__placeholder-column"
                      style="left: 399px;"
                    />
                    <div
                      class="grid-view__placeholder-column"
                      style="left: 599px;"
                    />
                  </div>
                   
                  <div
                    class="grid-view__rows"
                    style="transform: translateY(0px) translateX(0px);"
                  >
                    <div
                      class="grid-view__row"
                    >
                      <!---->
                       
                      <div
                        class="grid-view__column"
                        style="width: 200px;"
                      >
                        <div
                          class="grid-view__cell"
                        >
                          <div
                            class="grid-field-text"
                          >
                            name
                          </div>
                        </div>
                      </div>
                      <div
                        class="grid-view__column"
                        style="width: 200px;"
                      >
                        <div
                          class="grid-view__cell grid-field-long-text__cell"
                        >
                          <div
                            class="grid-field-long-text"
                          >
                            last_name
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                   
                  <!---->
                </div>
              </div>
               
              <div
                class="grid-view__foot"
              >
                <div
                  class="grid-view__foot-logo"
                >
                  <a
                    href="https://baserow.io"
                    target="_blank"
                    title="Baserow - open source no-code database tool and Airtable alternative"
                  >
                    <img
                      alt="Baserow - open source no-code database tool and Airtable alternative"
                      height="14"
                      src="@baserow/modules/core/static/img/logo.svg"
                    />
                  </a>
                </div>
                 
                <!---->
              </div>
            </div>
             
            <div
              style="display: none;"
            >
              <div
                class="grid-view__field-dragging"
                style="width: 0px; left: 0px;"
              />
               
              <div
                class="grid-view__field-target"
                style="left: 0px;"
              />
            </div>
          </div>
           
          <div
            class="grid-view__row-dragging-container"
            style="display: none;"
          >
            <div
              class="grid-view__row-dragging"
              style="width: 860px; top: 0px;"
            />
             
            <div
              class="grid-view__row-target"
              style="width: 860px; top: 0px;"
            />
          </div>
           
           
        </div>
         
        <!---->
      </div>
    </div>
  </div>
</div>
`;
